ads/google/a4a/utils.js

```diff
@@ -331,3 +335,46 @@ export function additionalDimensions(win, viewportSize) {
           innerWidth,
           innerHeight].join();
 };
+
+/**
+ * Creates amp-analytics element within a4a element using urls specified
+ * with amp-ad closest selector and min 50% visible for 1 sec.
+ * @param {!../../../extensions/amp-a4a/0.1/amp-a4a.AmpA4A} a4a
+ * @param {!../../../src/service/extensions-impl.Extensions} extensions
+ * @param {!Array<string>} urls
+ */
+export function injectActiveViewAmpAnalyticsElement(a4a, extensions, urls) {
+  dev().assert(urls.length);
+  extensions.loadExtension('amp-analytics');
+  const ampAnalyticsElem =
+    a4a.element.ownerDocument.createElement('amp-analytics');
+  const config = {
+    'transport': {'beacon': false, 'xhrpost': false},
+    'triggers': {
+      'continuousVisible': {
+        'on': 'visible',
+        'visibilitySpec': {
+          'selector': 'amp-ad',
+          'selectionMethod': 'closest',
+          'visiblePercentageMin': 50,
+          'continuousTimeMin': 1000,
+        },
+      },
+    },
+  };
+  const requests = {};
+  for (let idx = 1; idx <= urls.length; idx++) {
+    // TODO: Ensure url is valid and not freeform JS?
+    requests[`visibility${idx}`] = `${urls[idx - 1]}`;
+  }
+  // Security review needed here.
+  config['requests'] = requests;
+  config['triggers']['continuousVisible']['request'] = Object.keys(requests);
+  const scriptElem = createElementWithAttributes(
+      /** @type {!Document} */(a4a.element.ownerDocument), 'script', {
+        'type': 'application/json',
+      });
+  scriptElem.textContent = JSON.stringify(config);
+  ampAnalyticsElem.appendChild(scriptElem);
+  a4a.element.appendChild(ampAnalyticsElem);
```

Comment:
Is there any downside to having two `amp-analytics` elements inside the same `amp-ad` block, in case, say, the publisher has inserted one in the page source?  This code adds a second -- will that be problematic?